confluent_kafka/src/Consumer.c

```diff
@@ -508,6 +589,22 @@ static PyMethodDef Consumer_methods[] = {
 	  "  :raises: KafkaException\n"
 	  "\n"
 	},
+	{ "get_watermark_offsets", (PyCFunction)Consumer_get_watermark_offsets,
```

Comment:
re watermark consistency:
I briefly discussed such an API with @hachikuji about six months ago, but we didnt really decide on anything.
Prior to librdkafka 0.9.0 I added two APIs, query_watermark_offsets(ask broker) and get_watermark_offsets(cached) which the Python client abstracts in one call.
In that very limited sense it makes the API consistent.

But I agree with you that a more generic API that allows querying for any offset (logical, time based, absolute) makes more sense.

re caching: the main use of the cached API is to allow an application to check the currently known high watermark (and possibly somewhat outdated low watermark if stats is enabled) in its fast path with low cost. Typical use is an updated consumer lag (even though the stats already exposes that..).